src/drivers/QtTestDriver.cpp

```diff
@@ -8,25 +8,32 @@ namespace cucumber {
 namespace internal {
 
 const InvokeResult QtTestStep::invokeStepBody() {
-    QTemporaryFile file;
-    if (!file.open()) {
-        return InvokeResult::failure("Unable to open temporary file needed for this test");
+    QString file_name;
+    {
+        QTemporaryFile file;
+        if (!file.open()) {
+            return InvokeResult::failure("Unable to open temporary file needed for this test");
+        }
+        file.close();
     }
-    file.close();
 
-    QtTestObject testObject(this);
+    file_name += ".txt";
+
+    QtTestObject testObject( this );
     int returnValue = QTest::qExec(
         &testObject,
         QStringList() << "test"
-                      << "-o" << file.fileName()
+                      << "-o" << file_name
     );
-    if (returnValue == 0) {
-        return InvokeResult::success();
-    } else {
-        file.open();
-        QTextStream ts(&file);
-        return InvokeResult::failure(ts.readAll().toLocal8Bit());
+    auto ok = ( returnValue == 0 ) ? true : false;
+    std::string error_text;
+    QFile file( file_name );
+    if(!ok) {
+      file.open( QIODevice::ReadOnly );
+      error_text = QString::fromLocal8Bit( file.readAll() ).toStdString();
     }
+    //file.remove();
+    return ok ?  InvokeResult::success() :  InvokeResult::failure( error_text );
```

Comment:
Imho the readability of the old structure was better. Maybe something like:

```C++
  if(returnValue == 0) {
    return InvokeResult::success();
  }

  QFile file( file_name );
  file.open( QIODevice::ReadOnly );
  const auto error_text = QString::fromLocal8Bit( file.readAll() ).toStdString();

  return InvokeResult::failure( error_text );
}
```

Reply:
depends on the comment :smile: 
Currently the files (temporay) remain existing after test finished. This is no problem on linux where the /tmp/ is cleaned. on Windows those file fragments will remain. 
Should they be removed after test? 
[yes] -> no early exit or remove in early exit code
[no] -> then your suggestion is accepted.